models/modules/projected_d/projector.py

```diff
@@ -55,21 +72,56 @@ def calc_channels(pretrained, inp_res=224):
 
     return channels
 
-
-def _make_projector(im_res, cout, proj_type, expand=False):
+def create_timm_model(model_name,config_path,weight_path):
+    model = timm.create_model(model_name, pretrained=True)
+    return model
+
+def create_transformer_model(model_name,config_path,weight_path):
+    cfg = mmcv.Config.fromfile(config_path)
+    cfg.model.train_cfg = None
+    segformer = build_segmentor(cfg.model, train_cfg=None, test_cfg=cfg.get('test_cfg'))
+    load_checkpoint(
+        segformer, weight_path,
+        map_location='cpu',
+    )
+
+    model = segformer.backbone
+    
+    return model
+
+projector_models = {
+    "efficientnet": {
+        "model_name" : 'tf_efficientnet_lite0',
+        "create_model_function":create_timm_model,
+        "make_function" : _make_efficientnet
+    },
+    "transformer":{
+        "model_name" : '', #unused
+        "create_model_function":create_transformer_model,
+        "make_function" : _make_transformer
+    },
+
+}
+def _make_projector(projector_model,im_res, cout, proj_type, expand, config_path, weight_path):
     assert proj_type in [0, 1, 2], "Invalid projection type"
 
     ### Build pretrained feature network
-    model = timm.create_model('tf_efficientnet_lite0', pretrained=True)
-    pretrained = _make_efficientnet(model)
+    projector_gen=projector_models[projector_model]
+    model = projector_gen["create_model_function"](projector_gen["model_name"],config_path,weight_path)
+    pretrained = projector_gen['make_function'](model)
 
     # determine resolution of feature maps, this is later used to calculate the number
     # of down blocks in the discriminators. Interestingly, the best results are achieved
     # by fixing this to 256, ie., we use the same number of down blocks per discriminator
     # independent of the dataset resolution
-    im_res = 256
-    pretrained.RESOLUTIONS = [im_res//4, im_res//8, im_res//16, im_res//32]
-    pretrained.CHANNELS = calc_channels(pretrained)
+    
+    if projector_model != 'transformer' :
```

Comment:
We may want the opposite: the original projector is the default and the conditional is over the other positive alternative